library/core/src/hint.rs

```diff
@@ -121,7 +121,8 @@ pub fn black_box<T>(dummy: T) -> T {
     #[cfg(not(miri))] // This is just a hint, so it is fine to skip in Miri.
     // SAFETY: the inline assembly is a no-op.
     unsafe {
-        llvm_asm!("" : : "r"(&dummy));
+        // FIXME: Cannot use `asm!` because it doesn't support MIPS and other architectures.
+        llvm_asm!("" : : "r"(&mut dummy));
```

Comment:
Incidentally the old asm had bugs: you need `volatile` and `memory` clobbers to have the proper `black_box` effect. Currently rustc automatically adds `volatile` and LLVM adds `memory` when `volatile` is set, but we probably shouldn't rely on this.